luanaruiz9 tfds original data

luanaruiz9 opened the pull request:
Adding tensorflow dataset classes for original and iterative decoding productivity data (+ README).

luanaruiz9 committed:
Created README for generating tfds

luanaruiz9 committed:
Added tfds classes for original and iterative decoding productivity splits

luanaruiz9 committed:
Nits from previous (closed) pull request

santiaontanon-google commented on datasets/pcfg/tensorflow-datasets/README.md:
```diff
@@ -0,0 +1,13 @@
+## Creating TensorFlow datasets for PCFG
+
+To create TensorFlow datasets for each split of the PCFG data (iid, productivity and systematicity, either in the original or in the iterative decoding form), 
```
Small nit: about line breaks here. I would not insert any line breaks in long lines, or insert them using the standard 80 character line length.

santiaontanon-google commented on datasets/pcfg/tensorflow-datasets/pcfg_productivity_data.py:
```diff
@@ -0,0 +1,87 @@
+
+  def _split_generators(self, dl_manager: tfds.download.DownloadManager):
+    """Returns SplitGenerators."""
+    archive_path = dl_manager.manual_dir / 'data'
```
Oh, I am not familiar with this syntax, what does the "/" do here? is this an "append"?

luanaruiz9 replied:
This is an append which adds a backslash in between strings.

santiaontanon-google commented on datasets/pcfg/tensorflow-datasets/pcfg_productivity_data.py:
```diff
@@ -0,0 +1,87 @@
+"""pcfg_productivity_data dataset."""
```
It seems to me that the two dataset classes only differ in a few details (a couple of paths/filenames), right? Wouldn't it be better to then define a PcfgData class that has parameters in the constructor? and then the other datasets only require redefining the default constructor to call the PcfgData constructor with the right arguments, right? that would save 100s of lines of code. Let me know if it does not make sense :)

luanaruiz9 replied:
Because I am not sure if there is anyway to pass constructor attributes to the tfds command, I have taken a different approach by updating pcfg_productivity_data.py to generate both the original and the iterative decoding data. After running $ tfds build pcfg_productivity_data.py to build the dataset, the original data can be found in the 'train' and 'test' splits; and the iterative decoding data, in the 'it_dec_train', 'it_dec_val' and 'it_dec_test' splits. Per Josh's suggestion, the 'it_dec_val' set is used to assess the ability of the model to generalize in the standard sense (it has the same form as the 'it_dec_train' set); the 'it_dec_test' set is used to assess the ability of the model to generalize through iterative decoding. Let me know what you think!

luanaruiz9 committed:
Updated README.md

luanaruiz9 committed:
Updated pcfg_productivity_data.py

Updated pcfg_productivity_data.py to generate both the original and the iterative decoding data. After running $ tfds build pcfg_productivity_data.py to build the dataset, the original data can be found in the 'train' and 'test' splits; and the iterative decoding data, in the 'it_dec_train', 'it_dec_val' and 'it_dec_test' splits. Per Josh's suggestion, the 'it_dec_val' set is used to assess the ability of the model to generalize in the standard sense (it has the same form as the 'it_dec_train' set); the 'it_dec_test' set is used to assess the ability of the model to generalize through iterative decoding.

luanaruiz9 committed:
Deleted pcfg_productivity_data_original.py

Deleted pcfg_productivity_data_original.py as original productivity data is now generated in pcfg_productivity_data.py.

luanaruiz9 committed:
Updated data_generation.py to generate val set

Val dataset is a portion of the test set set aside to assess the ability of the model to generalize in the standard sense, i.e., to generalize to unseen data of the same form as the iterative decoding training data.

jtainslie commented on datasets/pcfg/data_generation.py:
```diff
@@ -139,22 +141,34 @@ def main(argv: Sequence[str]):
     for in_line, out_line in zip(test_input_lines, test_output_lines):
-      out_file_src.write(in_line)
-      out_file_tgt.write(out_line.strip('\n') + ' END\n')
+      if count < ratio*len(test_input_lines):
```
Nit: spacing.
```suggestion
      if count < ratio * len(test_input_lines):
```

jtainslie commented on datasets/pcfg/tensorflow-datasets/pcfg_productivity_data.py:
```diff
@@ -0,0 +1,104 @@
+    count = 0
+    for src, tgt in zip(source_lines,target_lines):
+      line_id = count
+      count+=1
```
Nit: spacing.
```suggestion
      count += 1
```

jtainslie commented on datasets/pcfg/tensorflow-datasets/pcfg_productivity_data.py:
```diff
@@ -0,0 +1,104 @@
+          source_path=extracted_path / 'it_dec_val.src',
+          target_path=extracted_path / 'it_dec_val.tgt',
+      ),
+      # val is the split used to check iterative decoding generalization
```
"test" instead of "val"?

jtainslie reviewed (approved):
Lgtm.

luanaruiz9 committed:
Corrected nit in productivity_data.py

luanaruiz9 committed:
Corrected nit in data_generation.py

luanaruiz9 committed:
Updated README.md to reflect changes in tfds class

Also corrected tfds build to include manual_dir argument.
